File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/tapdifferently/domain/TapDifferentlyObject.java
package com.zwstudio.logicpuzzlesandroid.puzzles.tapdifferently.domain;

public abstract class TapDifferentlyObject {
    public abstract String objTypeAsString();
    public static TapDifferentlyObject objTypeFromString(String str) {
        switch (str) {
            case "marker":
                return new TapDifferentlyMarkerObject();
            case "wall":
                return new TapDifferentlyWallObject();
            default:
                return new TapDifferentlyEmptyObject();
        }
    }
}

File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/tapdifferently/TapDifferentlyObject.kt
package com.zwstudio.logicpuzzlesandroid.puzzles.tapdifferently

import com.zwstudio.logicpuzzlesandroid.common.domain.Position

enum class TapDifferentlyObject {
    Empty, Hint, Marker, Wall
}

class TapDifferentlyGameMove(val p: Position, var obj: TapDifferentlyObject = TapDifferentlyObject.Empty)
